test/celeritas/optical/ScintillationGenerator.test.cc

```diff
@@ -0,0 +1,213 @@
+
+TEST_F(ScintillationGeneratorTest, basic)
+{
+    // Output data
+    std::vector<OpticalPrimary> storage(dist_.num_photons);
+
+    // Create the generator
+    ScintillationGenerator generate_photons(
+        dist_, params->host_ref(), make_span(storage));
+    RandomEngine& rng_engine = this->rng();
+
+    // Generate optical photons for a given input
+    auto photons = generate_photons(rng_engine);
+
+    // Check results
+    std::vector<real_type> energy;
+    std::vector<real_type> time;
+    std::vector<real_type> cos_theta;
+    std::vector<real_type> polarization_x;
+    std::vector<real_type> cos_polar;
+
+    for (auto i : range(dist_.num_photons))
+    {
+        energy.push_back(photons[i].energy.value());
+        time.push_back(photons[i].time);
+        cos_theta.push_back(
+            dot_product(photons[i].direction,
+                        dist_.points[StepPoint::post].pos
+                            - dist_.points[StepPoint::pre].pos));
+        polarization_x.push_back(photons[i].polarization[0]);
+        cos_polar.push_back(
+            dot_product(photons[i].polarization, photons[i].direction));
+    }
+
+    if (CELERITAS_REAL_TYPE == CELERITAS_REAL_TYPE_DOUBLE)
+    {
+        const real_type expected_energy[] = {9.3561354787881e-06,
+                                             9.39574581587642e-06,
+                                             1.09240249982534e-05,
+                                             6.16620934051192e-06};
+
+        const real_type expected_time[] = {7.30250028666843e-09,
+                                           1.05142594015847e-08,
+                                           3.11699961936832e-06,
+                                           2.68409417173788e-06};
+
+        const real_type expected_cos_theta[] = {0.937735542248463,
+                                                -0.775070967887639,
+                                                0.744857640134601,
+                                                -0.748206733055997};
+
+        const real_type expected_polarization_x[] = {-0.714016941727313,
+                                                     0.74609610658139,
+                                                     -0.456101107552679,
+                                                     0.0275013929040768};
+
+        const real_type expected_cos_polar[] = {0, 0, 0, 0};
+
+        EXPECT_VEC_SOFT_EQ(expected_energy, energy);
+        EXPECT_VEC_SOFT_EQ(expected_time, time);
+        EXPECT_VEC_SOFT_EQ(expected_cos_theta, cos_theta);
+        EXPECT_VEC_SOFT_EQ(expected_polarization_x, polarization_x);
+        EXPECT_VEC_SOFT_EQ(expected_cos_polar, cos_polar);
+    }
+}
+
+//---------------------------------------------------------------------------//
+TEST_F(ScintillationGeneratorTest, stress_test)
+{
+    // Generate a large number of optical photons
+    dist_.num_photons = 1234567;
```

Comment:
That is a large number indeed!!

Reply:
Reduce it to 123456 (~430ms --> 43ms for the stress test on my mac).